plugins/languages/treesitter/treesitter.nix

```diff
@@ -15,6 +15,58 @@ helpers.neovim-plugin.mkNeovimPlugin {
   description = ''
     Provides an interface to [tree-sitter]
 
+    ### Installing tree-sitter grammars from Nixpkgs
+
+    The default behavior of this module is to install every available grammar within the `nvim-treesitter` package.
+    You can control which ones get installed by passing the specific grammar packages you want:
+
+    ```nix
+      plugins.treesitter = {
+        enable = true;
+
+        grammarPackages = with pkgs.vimPlugins.nvim-treesitter.builtGrammars; [
+          bash
+          json
+          lua
+          make
+          markdown
+          nix
+          regex
+          toml
+          vim
+          vimdoc
+          xml
+          yaml
+        ];
+      };
+    ```
+
+    ### Installing tree-sitter grammars from nvim-treesitter
+
+    The default tree-sitter behavior is to not install any grammars through the plugin. You need to run `:TSInstall <grammar>`
+    or set the `plugins.treesitter.settings.ensure_installed` list to the grammars you want the plugin to fetch and install.
+
+    ```nix
+      plugins.treesitter = {
+        enable = true;
+
+        settings = {
+          # NOTE: You can set whether `nvim-treesitter` should automatically install the grammars.
+          auto_install = false;
+          ensure_installed = [
+            "git_config"
+            "git_rebase"
+            "gitattributes"
+            "gitcommit"
+            "gitignore"
+          ];
+        };
+      };
+    ```
+
+    NOTE: You can combine the functionality of `plugins.treesitter.nixGrammars` and `plugins.treesitter.settings.ensure_installed`
+    if there are cases that a grammar isn't available from nixpkgs or you'd prefer to have the plugin manage specific ones.
```

Comment:
```suggestion
    NOTE: You can combine the functionality of `plugins.treesitter.nixGrammars` and `plugins.treesitter.settings.ensure_installed`.
    This may be useful if a grammar isn't available from nixpkgs or you prefer to have specific grammars managed by nvim-treesitter.
```